### PR TITLE
[keycloakx] Replace postgres specific configuration with generic database configuration

### PR DESCRIPTION
**Pull request by PSanetra**
Replaces `.Values.postgresql` with `.Values.database`.

### CODE REVIEW

File: charts/keycloakx/templates/statefulset.yaml [hunk 1]
```diff
@@ -34,35 +34,32 @@ spec:
         {{- end }}
       labels:
         {{- include "keycloak.selectorLabels" . | nindent 8 }}
-        {{- if and .Values.postgresql.enabled (and .Values.postgresql.networkPolicy .Values.postgresql.networkPolicy.enabled) }}
-        {{ include "keycloak.postgresql.fullname" . }}-client: "true"
```
Comment (PSanetra) [inline charts/keycloakx/templates/statefulset.yaml:38]: I think configuring the network policy is now out of scope of this helm chart. Users should add a custom label for this purpose if they configure such a policy on the database helm chart.

Comment (thomasdarimont) [reply charts/keycloakx/templates/statefulset.yaml:38]: Yes this makes sense!

**Comment by thomasdarimont** [comment]
Thanks for this PR!

It seems that the CI build fails due some missing configuration:
https://github.com/thomasdarimont/helm-charts/runs/5882995208?check_suite_focus=true#step:9:270

File: charts/keycloakx/ci/h2-values.yaml [hunk 2]
```diff
@@ -4,47 +4,29 @@ command:
...
-    readOnly: true
-
-extraVolumes: |
-  - name: admin-creds
-    secret:
-      secretName: '{{ include "keycloak.fullname" . }}-admin-creds'
-
-postgresql:
-  enabled: false
+      user: keycloakadmin
```
Comment (thomasdarimont) [inline charts/keycloakx/ci/h2-values.yaml]: I think this should be just `admin` and `secret` to match with the values for the login test or ensure that the references are correct in https://github.com/codecentric/helm-charts/blob/master/charts/keycloak/templates/test/pod-test.yaml#L25

**Pull request merged**